File: api/src/test/java/net/blueshell/api/factory/model/EventPictureFactory.java
package net.blueshell.api.factory.model;

import com.github.javafaker.Faker;
import lombok.RequiredArgsConstructor;
import net.blueshell.api.model.event.EventPicture;
import org.springframework.stereotype.Component;

import java.util.concurrent.atomic.AtomicLong;

/**
 * Factory for EventPicture model test instances.
 */
@Component
@RequiredArgsConstructor
public class EventPictureFactory {

    private static final AtomicLong counter = new AtomicLong(1000);
    private final Faker faker;
    private final EventFactory eventFactory;
    private final FileFactory fileFactory;

    public EventPicture createBasic() {
        EventPicture ep = new EventPicture();
        ep.setId(generateId());
        ep.setEvent(eventFactory.createBasic());
        ep.setPicture(fileFactory.createImage());
        return ep;
    }

    public EventPicture createFull() {
        return createBasic();
    }

    public EventPicture createWithCustomizations(java.util.function.Consumer<EventPicture> customizer) {
        EventPicture ep = createFull();
        customizer.accept(ep);
        return ep;
    }

    private Long generateId() {
        return counter.incrementAndGet();
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/model/event/EventPictureFactory.kt
package net.blueshell.api.factory.model.event

import com.github.javafaker.Faker
import net.blueshell.api.factory.model.FileFactory
import net.blueshell.api.event.persistence.EventPicture
import org.springframework.stereotype.Component
import java.util.concurrent.atomic.AtomicLong
import java.util.function.Consumer

/**
 * Factory for EventPicture model test instances.
 */
@Component
class EventPictureFactory(
    private val faker: Faker,
    private val eventFactory: EventFactory,
    private val fileFactory: FileFactory
) {

    fun createBasic(): EventPicture {
        val picture = EventPicture()
        picture.event = eventFactory.createBasic()
        picture.picture = fileFactory.createImage()
        return picture
    }

    fun createFull(): EventPicture = createBasic()

    fun createWithCustomizations(customizer: Consumer<EventPicture>): EventPicture {
        val picture = createFull()
        customizer.accept(picture)
        return picture
    }

    private fun generateId(): Long = COUNTER.incrementAndGet()

    private companion object {
        val COUNTER = AtomicLong(1000)
    }
}
